dll/shellext/cabview/cabview.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * PROJECT:     ReactOS CabView Shell Extension
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     DLL entry point
+ * COPYRIGHT:   Copyright 2024 Whindmar Saksit <whindsaks@proton.me>
+ */
+
+#include "cabview.h"
+
+#include <initguid.h>
+DEFINE_GUID(CLSID_CabFolder, 0x0CD7A5C0,0x9F37,0x11CE,0xAE,0x65,0x08,0x00,0x2B,0x2E,0x12,0x62);
+
+class CCabViewModule : public CComModule { } g_Module;
+HINSTANCE g_hInst;
+
+BEGIN_OBJECT_MAP(ObjectMap)
+    OBJECT_ENTRY(CLSID_CabFolder, CCabFolder)
+END_OBJECT_MAP()
+
+EXTERN_C BOOL WINAPI DllMain(HINSTANCE hInstance, DWORD dwReason, LPVOID lpReserved)
+{
+    switch (dwReason)
+    {
+        case DLL_PROCESS_ATTACH:
+            DisableThreadLibraryCalls(hInstance);
+            g_hInst = hInstance;
+            g_Module.Init(ObjectMap, hInstance, NULL);
+            break;
+    }
+
+    return TRUE;
+}
+
+STDAPI DllCanUnloadNow()
+{
+    return g_Module.DllCanUnloadNow();
+}
+
+STDAPI DllGetClassObject(REFCLSID rclsid, REFIID riid, LPVOID *ppv)
+{
+    return g_Module.DllGetClassObject(rclsid, riid, ppv);
+}
+
+STDAPI DllRegisterServer()
+{
+    HRESULT hr;
+
+    hr = g_Module.DllRegisterServer(FALSE);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    hr = g_Module.UpdateRegistryFromResource(IDR_FOLDER, TRUE, NULL);
+    if (FAILED(hr))
```

Comment:
You could [ask yourself the same question](https://github.com/reactos/reactos/blame/3ecd2363a6d045a38aa68a1b5f17bb53ffaad3e4/dll/shellext/zipfldr/zipfldr.cpp#L149) because the DLL export functions and resources are based on your zipfldr.

Reply:
I wouldn't write the same code now indeed, this was a good 5 or 6 years ago :)